File: geometry/mylists.erl
-module(mylists).
-export([sum/1, qsort/1, pythag/1, perms/1, max/2]).

sum([H|T]) -> H + sum(T);
sum([]) -> 0.

qsort([]) -> [];
qsort([P|T]) ->
	qsort([X || X <- T, X < P])
	++ [P] ++
	qsort([X || X <- T, X >= P]).

pythag(N) ->
	[{A, B, C} ||
		A <- lists:seq(1, N),
		B <- lists:seq(1, N),
		C <- lists:seq(1, N),
		A + B > C, A + C > B, B + C > A,
		A*A + B*B =:= C*C
	].

perms([]) -> [[]];
perms(L) -> [[H|T] || H <- L, T <- perms(L--[H])].

max(X,Y) when X > Y -> X;
max(X,Y) -> Y.

File: geometry/mylists.py
def pythag(n):
    for n1 in xrange(1, n):
        for n2 in xrange(1, n):
            for n3 in xrange(1, n):
                if n1 + n2 > n3 and n1 + n3 > n2 and n2 +n3 > n1 and n1*n1 + n2*n2 == n3*n3:
                    print n1, n2, n3

def perms(L):
    n = []
    if L == []:
        return [n]
    for n1 in L[:]:
        n3 = L[:]
        n3.remove(n1)
        print 'n3', n3
        for n2 in perms(n3):
            n.append(n2)
    return n

if __name__ == "__main__":
    #pythag(80)
    print perms([1,2,3])
